### PR TITLE
CMIP6 YK summary notebook

### PR DESCRIPTION
**Pull request by kyleredilla**
This PR is for a notebook that performs the work needed to summarize monthly CMIP6 temperature and precipitation data for a small project with IARC researchers. It was included in the `ardac-toolbox` repo because we anticipate this kind of request becoming more common, and having some code to point to would be helpful!

To test, simply execute the new notebook. Check for typos / poor writing since this could end up as a public-facing notebook. Feel free to explore the resulting data products in any additional ways!

The additional files needed for testing can be found in `/beegfs/CMIP6/kmredilla/ardac_toolbox_PR21_files`

### CODE REVIEW

**Review by cstephen — CHANGES_REQUESTED** [review]
I gave this notebook a careful read and it all makes sense to me. Excellent/thorough work & lots of it!! I've added a handful of PR code comments/suggestions/questions.

Also:
- [x] `xarray`, `rioxarray`, `cftime`, and `rasterio` are missing from `environment.yml` and should maybe be added as part of this PR.
- [ ] I was not able to run the notebook because `YK_domain_YDNWR_plus_ecoregion_akalb83.shp` is not included in this PR. Not sure if this was intention or not, but I didn't see any comments/instructions for how/where to download it either. I'll try running the notebook again once I track down this file.

File: notebooks/cmip6_polygon_summary.ipynb [hunk 1]
```diff
@@ -0,0 +1,4226 @@
...
+    "# should be ~25 MB\n",
+    "r = requests.get(base_wcs_url.format(cov_id=\"cmip6_monthly_precipitation\", **bounds_di))\n",
+    "pr_ds = xr.open_dataset(io.BytesIO(r.content))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Unfortunately, the time cooridantes are incorrect. If we decode manually:"
```
[the remark this answers is not in the document]
Comment (kyleredilla) [reply notebooks/cmip6_polygon_summary.ipynb]: every time!

File: notebooks/cmip6_polygon_summary.ipynb [hunk 2]
```diff
@@ -0,0 +1,4226 @@
...
+   ],
+   "source": [
+    "yk_gdf.plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "There are multiple polygons in this shapefile, and we really just need one for summarization over the coarse ~100km grid of the CMIP6 data, so we'll have to do some prep work. \n",
```
[the remark this answers is not in the document]
Comment (kyleredilla) [reply notebooks/cmip6_polygon_summary.ipynb]: Good Q! fixed in [973255a](https://github.com/ua-snap/ardac-toolbox/pull/21/commits/973255a7c64afb15c194c052f0e28b809c2d3b0f)

**Comment by kyleredilla** [comment]
Hey thanks @cstephen for the review! Sorry for all the typos. Also the shapefile is in `/beegfs/CMIP6/kmredilla/ardac_toolbox_PR21_files` on Chinook, sorry I didn't make that more clear in the PR comment - let me know if you can't get to it.

File: notebooks/cmip6_polygon_summary.ipynb [hunk 3]
```diff
@@ -0,0 +1,4226 @@
...
+       "  padding: 0;\n",
+       "  margin: 0;\n",
+       "  float: left;\n",
+       "  padding-right: 10px;\n",
+       "  width: auto;\n",
+       "}\n",
+       "\n",
+       ".xr-attrs dt {\n",
+       "  font-weight: normal;\n",
+       "  grid-column: 1;\n",
```
[the remark this answers is not in the document]
Comment (cstephen) [reply notebooks/cmip6_polygon_summary.ipynb:4191]: Looks like the PR code suggestion formatting fell apart here, but this was just a minor grammar nitpick. I suggested changing:

> Now just write to a file for both of temperature and precipitation

To:

> Now just write to a file for both temperature and precipitation

[the remark this answers is not in the document]
Comment (kyleredilla) [reply notebooks/cmip6_polygon_summary.ipynb]: Ah okay! Fixed.

**Comment by cstephen** [comment]
Since this PR works great and it's only the conda `environment.yml` that has been giving me problems, I've reverted to the version of `environment.yml` from the `main` branch and captured the work left to do (+ some new tips) in issue #22.

**Pull request merged**